nodejs/eks/nodegroup.ts

```diff
@@ -1740,7 +1775,7 @@ function createManagedNodeGroupInternal(
     }
 
     let launchTemplate: aws.ec2.LaunchTemplate | undefined;
-    if (args.kubeletExtraArgs || args.bootstrapExtraArgs || args.enableIMDSv2) {
+    if (args.kubeletExtraArgs || args.bootstrapExtraArgs || args.enableIMDSv2 || args.gpu || args.amiId || args.amiType) {
```

Comment:
Why do we need to create a custom launch template if `amiType` is set? That's a valid input for `aws.eks.NodeGroup` on its own.

Aside from that, the list here gets a bit long now. What do you think about creating a method for deciding whether to create a custom launch template?